src/levelset/levelSetKernel.cpp

```diff
@@ -171,6 +171,79 @@ bool LevelSetKernel::intersectCellPlane( long id, const std::array<double,3> &ro
+ * @return the index corresponding to the given box face 
+ */
+int LevelSetKernel::computeBoxFaceIndex( long id, int intrLocalId, double tolerance) {
+
+    // Get cell and interface
+    const Cell &cell = m_mesh->getCell(id);
+    long intrGlobalId = cell.getInterfaces()[intrLocalId];
+    const Interface &interface = m_mesh->getInterface(intrGlobalId);
+
+    // Evaluate interface normal
+    ConstProxyVector<long> vertexIds = interface.getVertexIds();
+    std::size_t nVertices            = vertexIds.size();
+    BITPIT_CREATE_WORKSPACE(vertexCoordinates, std::array<double BITPIT_COMMA 3>, nVertices, ReferenceElementInfo::MAX_ELEM_VERTICES);
+    m_mesh->getVertexCoords(nVertices, vertexIds.data(), vertexCoordinates);
+
+    std::array<double, 3> intrNormal = interface.evalNormal(vertexCoordinates);
+
+    // Evaluate cell and interface centroids
+    std::array<double,3> cellCentroid      = m_mesh->evalElementCentroid(cell);
+    std::array<double,3> interfaceCentroid = m_mesh->evalElementCentroid(interface);
+
+    // Evaluate interface direction
+    double maxCoor = std::abs(intrNormal[0]);
+    int direction = 0;
+    for (int i = 1; i < 3; ++i) {
+        if (std::abs(intrNormal[i]) > maxCoor) {
+            maxCoor = std::abs(intrNormal[i]);
+            direction = i;
+        }
+    }
+    assert((utils::DoubleFloatingEqual()(maxCoor, 1.0, tolerance)));
+
+    // Check if interface is in the front side
+    bool frontFace = (interfaceCentroid[direction] > cellCentroid[direction]);
+
+    // Get box face id
+    return CGElem::getBoxFaceLocalIndex(direction, frontFace);
+}
+
+/*!
+ * Computes the part of the interface belonging to the subspace defined by the given plane.
+ * The normal to the plane points opposite to the subspace.
+ *
+ * @param[in] id is the index of cell
+ * @param[in] intrLocalId is the local index of the interface for the given cell
+ * @param[in] root is a point on the plane
+ * @param[in] normal is the normal of the plane
+ * @param[in] tolerance is the tolerance used for distance comparisons
+ * @param[out] interfaceCentroid is the centroid the intersected face computed as the arithmetic avretage of the face vertiecs
+ * @param[out] interfaceArea is the area of the intersected face
+ * @param[out] intersectionCentroid is the centroid of the intersection computed as the arithmetic avretage of the intersection points
+ * @return true if interface is intersected
+ */
+bool LevelSetKernel::intersectInterfacePlane( long id, int intrLocalId, const std::array<double,3> &root, const std::array<double,3> &normal, double tolerance, std::array<double, 3> *interfaceCentroid, double *interfaceArea, std::array<double, 3> *intersectionCentroid ) {
+
+    int boxFaceId = computeBoxFaceIndex(id, intrLocalId, tolerance);
+
+    std::array<double,3> minPoint;
+    std::array<double,3> maxPoint;
+    m_mesh->evalCellBoundingBox(id, &minPoint, &maxPoint);
```

Comment:
It seems it would be easier to just intersect the interface with the plane, without going through the bounding box. It just an intersection between a polygon and a plane, there may be better algorithm but a simple one should be: loop though the sides of the polygon, discard the sides that have both vertices below the plane, keep the sides that have both vertices above the plane, evaluate the intersection point if the side has one vertex above and one below the plane. Am I missing something?

Reply:
In CG there can be a set of function that does the cut: cutLine, cutTriangle, cutPolygon. These functions take as input the shape (line, triangle, polygon, ...) and the plane and return a list of points that define the shape after the cut.

Reply:
Done. I used already implemented functions whenever possible.